source/PlayerInfo.cpp

```diff
@@ -3523,6 +3573,39 @@ void PlayerInfo::Save(const string &path) const
 		mission.Save(out);
 	for(const Mission &mission : inactiveMissions)
 		mission.Save(out);
+	map<string, map<string, int>> offWorldMissionCargo;
+	map<string, map<string, int>> offWorldMissionPassengers;
+	for(const auto &it : ships)
+	{
+		const Ship &ship = *it.get();
+		// If the ship is at the player's planet, its mission cargo allocation does not need to be saved.
+		if(ship.GetPlanet() == planet)
+			continue;
+		for(const auto &cargo : ship.Cargo().MissionCargo())
+			offWorldMissionCargo[cargo.first->UUID().ToString()][ship.UUID().ToString()] = cargo.second;
+		for(const auto &passengers : ship.Cargo().PassengerList())
+			offWorldMissionPassengers[passengers.first->UUID().ToString()][ship.UUID().ToString()] = passengers.second;
```

Comment:
#5723 had a case for missions without UUIDs/with blank UUIDs. Should we not add missions to these lists in that case, meaning such cargo would fallback to teleporting to the player? Reason being that if we have multiple missions with blank UUIDs, then distributing the mission cargo back to ships on load will get messed up.

Reply:
Seems like missions without UUIDs get given ones, so we don't have to concern ourselves with dealing with missions without any.